File: app/src/main/java/com/anhhoang/popularmovies/ReviewsAdapter.java
package com.anhhoang.popularmovies;
/*
 * Copyright (C) 2013 The Android Open Source Project
 */


import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.anhhoang.popularmovies.model.Review;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by AnhHo on 3/21/2017.
 */

public class ReviewsAdapter extends RecyclerView.Adapter<ReviewsAdapter.ViewHolder> {
    private List<Review> mReviews;
    private int mTotalPages;
    private int mCurrentPage;
    private int mTotalReviews;

    public ReviewsAdapter(){
        mCurrentPage = 0;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        boolean attachToRoot = false;
        View view = LayoutInflater
                .from(parent.getContext())
                .inflate(R.layout.review_item_view, parent, attachToRoot);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Review review = mReviews.get(position);

        holder.mAuthorNameTv.setText(review.getAuthor());
        holder.mReviewTv.setText(review.getContent());
    }

    @Override
    public int getItemCount() {
        return mReviews != null ? mReviews.size() : 0;
    }

    public void setReviews(List<Review> reviews) {
        mReviews = reviews;
        notifyDataSetChanged();
    }

    public int getTotalPages() {
        return mTotalPages;
    }

    public void setTotalPages(int totalPages) {
        this.mTotalPages = totalPages;
    }

    public int getCurrentPage() {
        return mCurrentPage;
    }

    public void setCurrentPage(int currentPage) {
        this.mCurrentPage = currentPage;
    }

    public int getTotalReviews() {
        return mTotalReviews;
    }

    public void setTotalReviews(int totalReviews) {
        this.mTotalReviews = totalReviews;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.tv_author_name)
        public TextView mAuthorNameTv;
        @BindView(R.id.tv_review)
        public TextView mReviewTv;

        public ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: MovieApp/app/src/main/java/com/anhhoang/popularmovies/ReviewsAdapter.kt
package com.anhhoang.popularmovies

/*
 * Copyright (C) 2013 The Android Open Source Project
 */


import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.anhhoang.popularmovies.model.Review
import kotlinx.android.synthetic.main.review_item_view.view.*


/**
 * Created by AnhHo on 3/21/2017.
 */

class ReviewsAdapter : RecyclerView.Adapter<ReviewsAdapter.ViewHolder>() {
    var reviews: List<Review> = ArrayList()
        set(reviews) {
            field = reviews
            notifyDataSetChanged()
        }
    var totalPages: Int = 0
    var currentPage: Int = 0
    var totalReviews: Int = 0

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val attachToRoot = false
        val view = LayoutInflater
                .from(parent.context)
                .inflate(R.layout.review_item_view, parent, attachToRoot)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val review = reviews[position]

        holder.mAuthorNameTv.text = review.author
        holder.mReviewTv.text = review.content
    }

    override fun getItemCount(): Int {
        return reviews.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mAuthorNameTv = itemView.tv_author_name
        var mReviewTv = itemView.tv_review

    }
}
